editor/scene_tree_dock.cpp

```diff
@@ -2602,7 +2603,31 @@ void SceneTreeDock::_files_dropped(Vector<String> p_files, NodePath p_to, int p_
 void SceneTreeDock::_script_dropped(String p_file, NodePath p_to) {
 	Ref<Script> scr = ResourceLoader::load(p_file);
 	ERR_FAIL_COND(!scr.is_valid());
-	if (Node *n = get_node(p_to)) {
+	Node *n = get_node(p_to);
+
+	if (!n) {
+		return;
+	}
+
+	if (Input::get_singleton()->is_key_pressed(Key::CTRL)) {
+		Node *new_node = Object::cast_to<Node>(ClassDB::instantiate(scr->get_instance_base_type()));
+		ERR_FAIL_NULL(new_node);
```

Comment:
That's a potential memory leak? If the script's base type is not `Node`-derived / `RefCounted`-derived and the object is successfully instantiated then it will not be freed.
```suggestion
		StringName base_type = scr->get_instance_base_type();
		ERR_FAIL_NULL(!ClassDB::is_parent_class(base_type, "Node"));
		Node *new_node = Object::cast_to<Node>(ClassDB::instantiate(base_type));
		ERR_FAIL_NULL(new_node);
```